index.rst

```diff
@@ -1,61 +1,311 @@
+  - Consolidation of the data access interface, through butler gen3, for local data processing of corner wavefront sensors and remote access for full array mode.
+  - Consolidation of the OCS Controlled Pipeline System (`OCPS`_) for remote data processing.
+  - Conversion of the wavefront estimation pipeline (`WEP`_) into a butler gen3 pipeline task.
 
-     Caption text.
+.. _MTAOS: https://ts-mtaos.lsst.io/index.html
+.. _AOS Workshop: https://confluence.lsstcorp.org/x/zzTbAw
+.. _MTAOS (xml) interface: https://ts-xml.lsst.io/sal_interfaces/MTAOS.html
+.. _OCPS: https://dmtn-133.lsst.io
+.. _WEP: https://github.com/lsst-ts/ts_wep
 
-   Run: ``make html`` and ``open _build/html/index.html`` to preview your work.
-   See the README at https://github.com/lsst-sqre/lsst-technote-bootstrap or
-   this repo's README for more info.
+.. _Data-access-changes:
 
-   Feel free to delete this instructional comment.
+Data access changes
+===================
 
-:tocdepth: 1
+The consolidation of data access through butler gen3 represents an important change in the interface and operation of the MTAOS.
+The following is a series of improvements that can be implemented to improve the MTAOS usability.
 
-.. Please do not modify tocdepth; will be fixed when a new Sphinx theme is shipped.
+Processing calibration products
+-------------------------------
 
-.. sectnum::
+This is currently done via the ``processCalibrationProducts`` command, which expected a directory containing the raw calibration data.
+Instead, the butler instances hosting the data will also contain processed calibration products suitable for all exposures.
+This will have to be carefully managed by observatory personnel over time, but is a much more suitable approach than leaving the MTAOS in charge of the process.
 
-.. TODO: Delete the note below before merging new content to the master branch.
+Image ingestion
+---------------
 
-.. note::
+As part of the wavefront processing commands, the current version of the MTAOS receives files names and directory location.
+This data is initially ingested to a butler instance to allow them to be ISR corrected and later WEP processed.
+
+Since the data will now be ingested by a different component and will be made available to the MTAOS through butler, there is no more need for a local ingestion.
+
+Simplification of wfs process commands
+--------------------------------------
+
+As mention above, the wfs process commands receives file names and path.
+With the data already ingested in the butler the images can be uniquely identified by the ``visitId`` entry in the database.
+That means, all process commands need to be updated to receive the ``visitId`` instead of the files name and path.
+
+At the same time, most of the metadata that is now passed as part of the process commands (``fieldRA``, ``fieldDEC``, ``filter``, ``cameraRotation``) is also available as metadata in the butler and can be removed from the command payload.
```

Comment:
These informations are needed to decide the target of bright stars. The related calculation is unrelated to the butler. The reason to have this is to improve the performance of WEP. We could decide the targets based on bright star catalog without waiting for the ready of image data.

Reply:
Yes, but two things:

1. note that I am proposing to add a command specifically to do the target selection.
2. this information is already available in the butler as metadata. So there is no need to provide them, the software can programmatically get this information from a much more reliable source.

Reply:
The source selection happens before the butler is ready. We do not want to wait for the finish of butler ingestion to query the metadata to do the calculation of source section.

Reply:
I think TeWei's point is that, if `runWep` starts with source selection, and source selection doesn't have access to the butler metadata yet, then we still need these....

Reply:
well, in this case, the process would wait for the image to appear in the butler to retrieve the metadata. If you are in a hurry and want the source selection to be readily available you use the `selectSource` command. Otherwise you are just overloading the `runWEP` command.